### ISSUE TITLE
Scheduler importing is broken

### ISSUE THREAD

**Issue by Cpt0r** [mostjs-community/most-subject#29]
with `"most-subject": "^4.1.3",`
calling `next` on a `holdSubject` before an observer is attached produces an error because `this.scheduler` is pointing to a an improperly loaded module.

fixed by replacing (in SubjectSource.ts):
`const defaultScheduler = require('most/lib/scheduler/defaultScheduler');`
with:
`const defaultScheduler = require('most/lib/scheduler/defaultScheduler').default;`

At first I tried `import defaultScheduler from 'most/lib/scheduler/defaultScheduler';` but it complained it couldn't find the module...?

The reason this error doesn't happen with a non-holding stream is due to the `next` method only making the scheduler call after an observer is attached which replaces the broken scheduler with the one from higher up the call stack, while the holding subject calls the scheduler straight away to buffer itself:

HoldSubjectSource.ts:

```
  next (value: T) {
    if (this.scheduler === void 0) { return; }
    const time = this.scheduler.now();
    this.buffer = dropAndAppend({time, value}, this.buffer, this.bufferSize);

    if (this.active) {
      this._next(time, value);
    }
  }
```

SubjectSource.ts:

```
  next (value: T): void {
    if (!this.active || this.scheduler === void 0) return;
    this._next(this.scheduler.now(), value);
  }

  run (sink: Sink<T>, scheduler: Scheduler): Disposable<T> {
    const n = this.add(sink);
    if (n === 1) {
      this.scheduler = scheduler;
      this.active = true;
    }
    return new SubjectDisposable<T>(this, sink);
  }
```

**Comment by TylorS** [comment]
Ahh yes, this is a recent change in most.js proper to ES2015 modules instead of commonjs. These modules are not part of the public api, which is why it's still covered by the versions in this projects package.json and was not deemed a breaking change. These will need to be updated

**Comment by TylorS** [comment]
Temporarily if you downgrade most.js to v1.0.1 things will work, nothing else has happened in v1.0.2 or v1.0.3

**Comment by Cpt0r** [comment]
Confirmed, thanks.

**Comment by TylorS** [comment]
I think this should be solved now with version 5.0.0

**Comment by TylorS** [comment]
Closing this for now, feel free to take a look at the newest version, and reopen this if you encounter issues.
